tools/cli/domainCommands.go

```diff
@@ -902,16 +904,67 @@ func (d *domainCLIImpl) migrationDynamicConfigCheck(c *cli.Context) DomainMigrat
 			if currResp.Value != newResp.Value {
 				check = false
 				mismatchedConfigs = append(mismatchedConfigs, MismatchedDynamicConfig{
-					CurrResp: currResp,
-					NewResp:  newResp,
+					Key:        configKey,
+					CurrValues: []*types.DynamicConfigValue{},
+					NewValues:  []*types.DynamicConfigValue{},
 				})
 			}
 
-		} else {
-			// TODO: Implement for domainName + taskList + taskType filters case
+		} else if containsFilter(configKey, dc.DomainName.String()) && containsFilter(configKey, dc.TaskListName.String()) {
+			taskLists := c.StringSlice(FlagTaskList)
+			for _, taskList := range taskLists {
+
+				currRequest := dynamicconfig.ToGetDynamicConfigFilterRequest(configKey.String(), []dynamicconfig.FilterOption{
+					dynamicconfig.DomainFilter(domain),
+					dynamicconfig.TaskListFilter(taskList),
+				})
+
+				newRequest := dynamicconfig.ToGetDynamicConfigFilterRequest(configKey.String(), []dynamicconfig.FilterOption{
+					dynamicconfig.DomainFilter(newDomain),
+					dynamicconfig.TaskListFilter(taskList),
+				})
+
+				currResp, err := d.frontendAdminClient.GetDynamicConfig(ctx, currRequest)
+				if err != nil {
+					currResp = &types.GetDynamicConfigResponse{}
+				}
+				newResp, err := d.destinationAdminClient.GetDynamicConfig(ctx, newRequest)
+				if err != nil {
+					newResp = &types.GetDynamicConfigResponse{}
+				}
+
+				if currResp.Value != newResp.Value {
+					check = false
+
+					mismatchedConfigs = append(mismatchedConfigs, MismatchedDynamicConfig{
+						Key: configKey,
+						CurrValues: []*types.DynamicConfigValue{
+							&types.DynamicConfigValue{
+								Filters: []*types.DynamicConfigFilter{
+									&types.DynamicConfigFilter{
```

Comment:
you need to pass value as well https://sourcegraph.com/github.com/shijiesheng/cadence/-/blob/common/dynamicconfig/configstore/config_store_client_test.go?L106#L103:67

You can add a helper mapper function in this file `toDyanmicConfigValue(key dynamicconfig.Key, filters []dynamicconfig.Filter, filterValues []*Datablob, keyValue *DataBlob)`